Check if enmity list is empty for unclaimed mobs

cocosolos opened the pull request:
Prior to the recent changes the attack round would simply set the owner instead of checking enmity since without unclaiming mechanics it didn't really matter. Attack enmity is set in takePhysicalDamage, so missed attacks and 0 damage attacks didn't generate any enmity, BUT since the owner was getting set either way, during the mob roam tick it checks if it has anyone on their enmity list, and if it doesn't but it does have an owner, it adds base enmity there. This fixes an oversight in the claim code for when claim is attempted with an empty enmity list.

<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

cocosolos committed:
Check if enmity list is empty for unclaimed mobs

cocosolos committed:
Merge branch 'claim' into claim-miss-enmity

cocosolos committed:
Apply enmity on mobskill use and when claiming

Ensures that missed attacks still add the entity to the enmity list of the
defender so that claim attempt can proceed.

cocosolos committed:
Trigger declaim when disengaging due to range

ibm2431 commented on src/map/ai/states/mobskill_state.cpp:
```diff
@@ -102,6 +103,11 @@ bool CMobSkillState::Update(time_point tick)
     if (IsCompleted() && tick > m_finishTime)
     {
+        auto PTarget = GetTarget();
+        if (PTarget && PTarget->objtype == TYPE_MOB)
+        {
+            static_cast<CMobEntity*>(PTarget)->PEnmityContainer->UpdateEnmity(m_PEntity, 0, 0, true);
```
Will we run into any surprises if mobs of the same allegiance use buffing mobskills on each other (or a goblin drops his bomb near his buds)?

ibm2431 reviewed (commented):
Helps if I press the "finish your review" button!

cocosolos committed:
Adjust enmity add for mobskills

ibm2431 reviewed (approved):
Thanks as always, cocosolos!

cocosolos committed:
Make sure enmity isn't added when passing claim

Enmity/dirtying exp shouldn't be applied if a player not already on the enmity
list gets passed claim.
